### PR TITLE
Migrate from plenary to nvim_win_open

### PR DESCRIPTION
**Pull request by EpsilonKu**

### CODE REVIEW

File: lua/spectre/ui.lua [hunk 1]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
+
+    local help_win = vim.api.nvim_open_win(bufnr, true, {
+        title = title,
+        title_pos = 'center',
+        relative = 'cursor',
+        width = win_width + 3,
+        height = win_height + 2,
+        col = 4,
+        row = 4,
+        border = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" }
```
Comment (Subjective) [inline lua/spectre/ui.lua:164]:
```suggestion
        border = "rounded"
```

File: lua/spectre/ui.lua [hunk 2]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
-        border = true,
-        borderchars ={ "─", "│", "─", "│", "╭", "╮", "╯", "╰" },
+
+    local check = false
+    local bufnr = vim.api.nvim_create_buf(false, true)
+    api.nvim_buf_set_option(bufnr, "bufhidden", "wipe")
+    api.nvim_buf_set_lines(bufnr, 0, -1, true, content)
+
+    local help_win = vim.api.nvim_open_win(bufnr, true, {
+        title = title,
```
Comment (Subjective) [inline lua/spectre/ui.lua:157]: imo it looks a lot better with padding
```suggestion
        title = " " .. title .. " ",
```

File: lua/spectre/ui.lua [hunk 3]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
+        title_pos = 'center',
+        relative = 'cursor',
+        width = win_width + 3,
+        height = win_height + 2,
+        col = 4,
+        row = 4,
```
Comment (Subjective) [inline lua/spectre/ui.lua:163]: This positioning is more consistent with lsp hover:
```suggestion
        col = 0,
        row = 1,
```

File: lua/spectre/ui.lua [hunk 4]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
+    local bufnr = vim.api.nvim_create_buf(false, true)
+    api.nvim_buf_set_option(bufnr, "bufhidden", "wipe")
+    api.nvim_buf_set_lines(bufnr, 0, -1, true, content)
+
+    local help_win = vim.api.nvim_open_win(bufnr, true, {
+        title = title,
+        title_pos = 'center',
+        relative = 'cursor',
+        width = win_width + 3,
+        height = win_height + 2,
```
Comment (Subjective) [inline lua/spectre/ui.lua:161]: opinionated, but I don't think there should be additional right/bottom padding
```suggestion
        width = win_width,
        height = win_height,
```

File: lua/spectre/ui.lua (hunk 2 above)
Comment (Subjective) [inline lua/spectre/ui.lua:157]: This disables line number and relative line number, among other options
```suggestion
        style = "minimal",
        title = title,
```

File: lua/spectre/ui.lua [hunk 5]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
+        col = 4,
+        row = 4,
+        border = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" }
     })
+    api.nvim_win_set_option(help_win, 'winblend', 0)
+    api.nvim_win_set_option(help_win, 'number', false)
```
Comment (Subjective) [inline lua/spectre/ui.lua:167]: unecessary when setting `style = "minimal"`
```suggestion
```

File: lua/spectre/ui.lua [hunk 6]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
-        padding = { 1, 1, 1, 1 },
-        enter   = false,
-        width   = win_width + 2,
-        height  = win_height + 2,
-        col     = "cursor+2",
-        line    = "cursor+2",
-        border = true,
-        borderchars ={ "─", "│", "─", "│", "╭", "╮", "╯", "╰" },
+
+    local check = false
```
Comment (Subjective) [inline lua/spectre/ui.lua:151]: unused variable
```suggestion
```

File: lua/spectre/ui.lua [hunk 7]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
+    local bufnr = vim.api.nvim_create_buf(false, true)
+    api.nvim_buf_set_option(bufnr, "bufhidden", "wipe")
+    api.nvim_buf_set_lines(bufnr, 0, -1, true, content)
+
+    local help_win = vim.api.nvim_open_win(bufnr, true, {
```
Comment (Subjective) [inline lua/spectre/ui.lua:156]: I don't think we should automatically enter the help window, imo it should be similar to lsp hover where the window is first displayed, then hitting the keymap again enters it
```suggestion
    local help_win = vim.api.nvim_open_win(bufnr, false, {
```

File: lua/spectre/ui.lua [hunk 8]
```diff
@@ -142,36 +141,37 @@ function M.render_header(opts)
...
+        border = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" }
     })
+    api.nvim_win_set_option(help_win, 'winblend', 0)
+    api.nvim_win_set_option(help_win, 'number', false)
+    api.nvim_buf_set_keymap(bufnr, 'n', '<Esc>','<CMD>lua vim.api.nvim_win_close(' .. help_win  .. ', true)<CR>', {noremap = true})
 
-    vim.api.nvim_win_set_option(help_win, 'winblend', 0)
     api.nvim_create_autocmd({
-        'CursorMoved',
         'CursorMovedI',
```
Comment (Subjective) [inline lua/spectre/ui.lua:171]: These should be added back to close the window if we disable automatically entering the help window (`vim.api.nvim_open_win(bufnr, false...`)
```suggestion
        'CursorMoved',
        'CursorMovedI',
        'BufHidden',
        'BufLeave',
        'InsertEnter',
        'WinScrolled',
        'BufDelete',
```

**Review by Subjective — CHANGES_REQUESTED** [review]

**Review by Subjective — CHANGES_REQUESTED** [review]
Also we should probably rename "Mapping" to "Mappings"

File: lua/spectre/ui.lua (hunk 8 above)
Comment (EpsilonKu) [reply lua/spectre/ui.lua:171]: It won't if it focus on window. I will add it back so it won't focus on it.

**Comment by EpsilonKu** [comment]
@Subjective what about now?

**Comment by Subjective** [comment]
@EpsilonKu LGTM!

**Pull request merged**